File: backend/src/main/scala/domains/problem/model/ProblemData.scala
package domains.problem.model



import io.circe.{Decoder, Encoder}

final case class ProblemData(value: Option[ProblemDataFilename])

object ProblemData:
  def parse(raw: Option[String]): Either[String, ProblemData] =
    raw match
      case None => Right(ProblemData(None))
      case Some(value) =>
        val normalized = value.trim
        if normalized.isEmpty then Right(ProblemData(None))
        else ProblemDataFilename.parse(normalized).map(filename => ProblemData(Some(filename)))

  given Encoder[ProblemData] = Encoder.encodeOption[ProblemDataFilename].contramap(_.value)
  given Decoder[ProblemData] = Decoder.decodeOption[String].emap(parse)

File: frontend/src/objects/problem/ProblemData.ts
import type { ProblemDataFilename } from '@/objects/problem/ProblemDataFilename'

/** 题目数据入口信息；value 为空表示尚未上传或未绑定数据包。 */
export type ProblemData = {
  value: ProblemDataFilename | null
}
